composable_prepare_nic_configs.yml

```diff
@@ -38,6 +38,24 @@
             with_items:
               - "{{ product_name.results }}"
 
+          - name: Get rack of r640s
+            set_fact: 
+              rack: "{{ rack|default[] + [item.1.stdout.split('-')[0]] }}"
+            when: item.0 == "r640"
+            with_together:
+              - "{{ machine_types }}"
+              - "{{ host_list.results }}"
+            
+          - name: set fact rhel8_interfaces (r640)
+            set_fact:
+              rhel8_interfaces: "{{  rhel8_interfaces|default([]) + [ (rack == 'f04') | ternary('rhel8_interfaces_f04', 'rhel8_interfaces_other') ] }}"
```

Comment:
rhel8_interfaces is a list of interfaces for all heterogeneous machines when composable_roles is enabled. Please adjust the task so that it is applicable for all cases 
1) Detect interfaces for 1029u
2) detect interfaces for r640 and
3) remaining machines